targets/TARGET_NORDIC/TARGET_NRF5x/qspi_api.c

```diff
@@ -400,6 +417,104 @@ nrf_qspi_frequency_t nrf_frequency(int hz)
     return freq;
 }
 
+// Private helper to read nRF5x SFDP data using QSPI
+qspi_status_t sfdp_read(qspi_t *obj, const qspi_command_t *command, void *data, size_t *length)
+{
+    ret_code_t ret_code;
+    static bool b_init = false;
+    static uint8_t sfdp_rx[SFDP_DATA_LEN] = { 0 };
+
+    if (b_init == false) {
+        // get the SFDP data usig nRF5x QSPI custom instuctions and long frame mode (lfen)
+        nrf_qspi_cinstr_conf_t qspi_cinstr_config = { 0 };
+        qspi_cinstr_config.opcode = command->instruction.value;
+        qspi_cinstr_config.io2_level = true;
+        qspi_cinstr_config.io3_level = true;
+        qspi_cinstr_config.wipwait   = false;
+        qspi_cinstr_config.wren      = false;
+        qspi_cinstr_config.lfen      = true;
+        qspi_cinstr_config.lfstop    = false;
+
+        // read the SFDP data, cmd + 8 bytes at a time
+        qspi_cinstr_config.length = NRF_QSPI_CINSTR_LEN_9B;
+        for (uint32_t i = 0; i < SFDP_READ_MAX; ++i) {
+
+            static uint32_t rx_i = 0;
+            uint8_t sfdp_data[SFDP_READ_LEN] = { 0 };
```

Comment:
will the compiler know not to reallocate it each iteration, and in the following loop (line 477) ?
it may be better performance to allocate it outside the loop and just memset it here.

Reply:
This makes sense. I've made the requested changes.